packages/web3/src/contract/contract.ts

```diff
@@ -1758,6 +1773,19 @@ export interface CallContractResult<R> {
   debugMessages: DebugMessage[]
 }
 
+export interface SignExecuteContractMethodParams<T extends Arguments = Arguments> {
+  args: T
+  signer: SignerProvider
+  attoAlphAmount?: Number256
+  tokens?: Token[]
+  approve?: {
+    attoAlphAmount?: Number256
+    tokens?: Token[]
+  }
```

Comment:
Question: is it possible to only use one `attoAlphAmount` and `tokens` arg?

Reply:
Great question. I am hesitating on this as well. 
For `attoAlphAmount`, we often include the extra amount for gas, so we can't automatically approve everything. 
For `tokens`, maybe we can automatically approve everything, but there are also scenarios where users just want to provide tokens in the transaction inputs without spending it (e.g. an NFT holder should be able to do something in a smart contract without transferring it)

But i do feel it's a bit duplicated to have both `approve` and `attoAlphAmount` & `tokens`, any suggestions for better interface is much appreciated.

Reply:
It seems like users don't need to consider the gas fee. The full node will select utxos for `attoAlphAmount + gasFee`: https://github.com/alephium/alephium/blob/master/flow/src/main/scala/org/alephium/flow/core/UtxoSelectionAlgo.scala#L146
but as you said, sometimes users need to approve extra dust amount, this means the script will approve more ALPH than before.

Reply:
I suppose one approach is to default the "approve" value the `attoAlphAmount` and `tokens` amount, so in most cases we don't need to specify `approve`. But if we want to be explicit we have the option to do that. WDTY?

Reply:
That sounds great 👍

Reply:
Let's keep things straightforward and stick with attoAlphAmount and tokens for now. Approving a bit more dustAmount shouldn't cause any harm. When building the transaction, attoAlphAmount and tokens indicate how much of the assets the user wants to use, not how much is ultimately used in the transaction.

I recommend removing `approve` entirely for now and only adding it later if necessary.

@h0ngcha0, can you confirm that gas fee is not included in attoAlphAmount? It should not be, as including it would make things too complicated.

Reply:
Yeah, gas fee is not included in `attoAlphAmount`.
Will remove `approve` now :+1: